### PR TITLE
Bluetooth: BAP: Shell: Add endpoint state info

### LINKED ISSUE #70838: LE Audio: `bap list` should also show the states of each stream/endpoint
**Is your enhancement proposal related to a problem? Please describe.**
The `bap list` command is a nice way to get an overview of the existing streams and endpoints, but it does not show the states of them, which makes it hard to tell what commands can be done on which streams. 

**Describe the solution you'd like**
The command should use the `bt_bap_ep_get_info` to get the state of each stream/endpoint and show that as well. 

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by jthm-ot**
The bap list command now also shows the states of each endpoint.

Fixes #70838

### CODE REVIEW

File: subsys/bluetooth/audio/shell/bap.c [hunk 1]
```diff
@@ -1631,7 +1631,15 @@ static void conn_list_eps(struct bt_conn *conn, void *data)
...
-			shell_print(sh, "    #%u: ep %p", i, ep);
+			struct bt_bap_ep_info ep_info;
+			int err;
+
+			err = bt_bap_ep_get_info(ep, &ep_info);
+			if (err == 0) {
+				shell_print(sh, "    #%u: ep %p (state: %d)", i, ep, ep_info.state);
+			} else {
+				shell_print(sh, "    #%u: ep %p (get ep info err: %d)", i, ep, err);
+			}
```
Comment (Thalley) [inline subsys/bluetooth/audio/shell/bap.c:1642]: I would probably just omit the error cases. 

If you want to keep them, we should use `shell_error` instead. 

The reason why I'd omit them, is that the function can only fail with invalid input, and we don't supply any invalid input here.

**Review by Thalley — CHANGES_REQUESTED** [review]
Minor request, otherwise LGTM

File: subsys/bluetooth/audio/shell/bap.c (hunk 1 above)
Comment (jthm-ot) [reply subsys/bluetooth/audio/shell/bap.c:1642]: Ok. I will omit the error case then

**Review by Thalley — APPROVED** [review]
LGTM. 

We could consider moving `bt_bap_pa_state_str` to a public API and use that here, so that we can get the string representations of the states

**Review by larsgk — APPROVED** [review]

**Pull request merged**